Packs/GoogleVertexAI/Integrations/GoogleVertexAI/GoogleVertexAI.py

```diff
@@ -0,0 +1,308 @@
+import demistomock as demisto  # noqa: F401
+from CommonServerPython import *  # noqa: F401
+
+
+''' IMPORTS '''
+
+
+import json
+import urllib3
+
+# Disable insecure warnings
+urllib3.disable_warnings()
+
+
+''' GLOBAL Variables '''
+
+
+DISABLE_SSL = demisto.params().get('insecure', False)
+PROXY = demisto.params().get('proxy')
+PROMPT = demisto.params().get('prompt')
+SERVICE_SCOPES = "https://www.googleapis.com/auth/cloud-platform"
+REDIRECT_URI = 'https://oproxy.demisto.ninja/authcode'
+AUTH_URL = 'https://accounts.google.com/o/oauth2/auth'
+ACCESS_TOKEN_URL= 'https://oauth2.googleapis.com/token'
+PROJECT_ID = demisto.params().get('ProjectID')
+URL = 'https://us-central1-aiplatform.googleapis.com/v1/projects/'
+AI_Model = 'chat-bison:predict'
+TOKEN = demisto.params().get('token')
+CLIENT_ID = demisto.params().get('ID')
+CLIENT_SECRET = demisto.params().get('Secret')
+AUTH_CODE = demisto.params().get('Authentication_Code')
+
+
+''' CLIENT CLASS '''
+
+
+class Client(BaseClient):
+        """ Client class to interact with Google Vertex AI API
+        """
+
+        def __init__(self, token_str: str, base_url: str, proxy: bool, verify: bool):
+            super().__init__(base_url=URL, proxy=PROXY, verify=verify)
+            self.token_str = token_str
+            self.base_url = base_url
+            self.headers = {'Authorization': f"Bearer {self.token_str}", "Content-Type": "application/json"}
+
+        def PaLMModel(self, prompt: str):
+            options = {"instances": [{"messages": [{"content": prompt}]}]}
+            return self._http_request(method='POST', url_suffix=f'{PROJECT_ID}/locations/us-central1/publishers/google/models/{AI_Model}', json_data=options, headers=self.headers)
+
+        
+''' MAIN FUNCTIONS '''
+
+
+def createAuthorizationURL():
+    # The client ID and access scopes are required.
+    authorization_url = f"{AUTH_URL}/oauthchooseaccount?scope={SERVICE_SCOPES}&access_type=offline&prompt=consent&response_type=code&state=state_parameter_passthrough_value&redirect_uri={REDIRECT_URI}&client_id={CLIENT_ID}"
+    return authorization_url
+
+
+def check_access_token_validation():
+    # Access tokens expires in 1 hour, then using refresh_token from test_module() function we will request for a new refresh token
+
+    integration_context: dict = get_integration_context()
+    access_token: str = integration_context.get('access_token', '')
+    valid_until: int = integration_context.get('valid_until', int)
+    time_now = epoch_seconds()
+    if len(access_token) > 0:
```

Comment:
I think one of the if statement on access_token is redundant. 
if len(access_token) > 0 ( means not empty None is also considered empty)
if access_token .. also means access_token is not empty or True

perhaps if access_token and (time_now < valid_until ): 
can do the trick in a single if statement adding to readability and less indentation.

Reply:
@thefrieddan1 done thank you for your feedback